src/components/calcite-picker/calcite-picker.tsx

```diff
@@ -0,0 +1,226 @@
+
+  /**
+   * Multpile Works similar to standard radio buttons and checkboxes.
+   * It also affects the presented icon when in Selection mode.
+   * When true, a user can select multiple items at a time.
+   * When false, only a single item can be selected at a time,
+   * When false, selecting a new item will deselect any other selected items.
+   */
+  @Prop({ reflect: true }) multiple = false;
+
+  /**
+   * The heading label for the entire Picker.
+   * Not to be confused with the heading for an individual item or for a sub-group of items.
+   */
+  @Prop({ reflect: true }) textHeading: string;
+
+  // --------------------------------------------------------------------------
+  //
+  //  Private Properties
+  //
+  // --------------------------------------------------------------------------
+
+  @State() selectedValues: Map<string, HTMLCalcitePickerItemElement> = new Map();
+
+  items: HTMLCalcitePickerItemElement[];
+
+  lastSelectedItem: HTMLCalcitePickerItemElement = null;
+
+  guid = `calcite-picker-${guid()}`;
+
+  observer = new MutationObserver(() => this.setupItems());
+
+  // --------------------------------------------------------------------------
+  //
+  //  Private Properties
+  //
+  // --------------------------------------------------------------------------
+
+  @Element() el: HTMLElement;
+
+  // --------------------------------------------------------------------------
+  //
+  //  Lifecycle
+  //
+  // --------------------------------------------------------------------------
+
+  connectedCallback() {
+    this.setupItems();
+  }
+
+  componentDidLoad() {
+    this.observer.observe(this.el, { childList: true, subtree: true });
+  }
+
+  componentDidUnload() {
+    this.observer.disconnect();
+  }
+
+  // --------------------------------------------------------------------------
+  //
+  //  Events
+  //
+  // --------------------------------------------------------------------------
+
+  @Event() calcitePickerSelectionChange: EventEmitter;
+
+  @Listen("calcitePickerItemToggle") calcitePickerItemToggleHandler(event) {
+    event.stopPropagation(); // private event
+    const { selectedValues } = this;
+    const { item, value, selected, shiftPressed } = event.detail;
+    if (selected) {
+      if (!this.multiple) {
+        this.deselectSiblingItems(item);
+      }
+      if (this.multiple && shiftPressed) {
+        this.selectSiblings(item);
+      }
+      selectedValues.set(value, item);
+    } else {
+      selectedValues.delete(value);
+    }
+    this.lastSelectedItem = item;
+    this.calcitePickerSelectionChange.emit(selectedValues);
+  }
+
+  // --------------------------------------------------------------------------
+  //
+  //  Private Methods
+  //
+  // --------------------------------------------------------------------------
+
+  setupItems(): void {
+    this.items = Array.from(this.el.querySelectorAll("calcite-picker-item"));
+    this.items.forEach((item) => {
```

Comment:
could the onclick for shift key handling be setup here instead?

Reply:
I looked into that. the entire click handler could be moved here (shift and all). But that didn't solve any problems, it just moved them. The state/prop split is what solved the problems and provided a solution for introducing watching the "selected" prop. I don't see any reason to move the click since it won't be solving anything.